### PR TITLE
[ICDS Dashboard] generation of dashboard usage report

### PR DESCRIPTION
**Pull request by challabeehyv**
https://app.asana.com/0/0/1137448610458201/f

##### SUMMARY
Added the code for generation of dashboard usage report

##### FEATURE FLAG
No

### CODE REVIEW

File: custom/icds_reports/sqldata/exports/dashboard_usage.py
Comment (Rohit25negi) [inline custom/icds_reports/sqldata/exports/dashboard_usage.py]: What will happen is the user is district level and it will have 'All' as its value. Same if a user is Staet level User.

**Review by Rohit25negi — CHANGES_REQUESTED** [review]
hey, Left some comments.

File: custom/icds_reports/sqldata/exports/dashboard_usage.py
Comment (challabeehyv) [reply custom/icds_reports/sqldata/exports/dashboard_usage.py]: done

File: custom/icds_reports/sqldata/exports/dashboard_usage.py
Comment (challabeehyv) [reply custom/icds_reports/sqldata/exports/dashboard_usage.py]: i will exclude present date

File: custom/icds_reports/sqldata/exports/dashboard_usage.py
Comment (challabeehyv) [reply custom/icds_reports/sqldata/exports/dashboard_usage.py]: intially we were getting these values with aggregation level 5. So there was no chance of all. Since we have moved to new implementation, i have changed accordingly

File: custom/icds_reports/sqldata/exports/dashboard_usage.py
Comment (challabeehyv) [reply custom/icds_reports/sqldata/exports/dashboard_usage.py]: changed

File: custom/icds_reports/sqldata/exports/dashboard_usage.py
Comment (Rohit25negi) [inline custom/icds_reports/sqldata/exports/dashboard_usage.py]: this works for both. WebUser and CommcareUser?

File: custom/icds_reports/sqldata/exports/dashboard_usage.py
Comment (Rohit25negi) [inline custom/icds_reports/sqldata/exports/dashboard_usage.py]: it will contain info like 'state_id'.
nit: probably something like `user_loc_id_key` would be a better name.

File: custom/icds_reports/sqldata/exports/dashboard_usage.py
Comment (Rohit25negi) [inline custom/icds_reports/sqldata/exports/dashboard_usage.py]: this function returns the values but its not receiving it here. IS that returning value still needed?

File: custom/icds_reports/sqldata/exports/dashboard_usage.py
Comment (Rohit25negi) [inline custom/icds_reports/sqldata/exports/dashboard_usage.py]: I hope you tested this that it is going back in days if today's table is not present

File: custom/icds_reports/sqldata/exports/dashboard_usage.py
Comment (Rohit25negi) [inline custom/icds_reports/sqldata/exports/dashboard_usage.py]: good to also describe its excluding current day.

File: custom/icds_reports/sqldata/exports/dashboard_usage.py
Comment (Rohit25negi) [inline custom/icds_reports/sqldata/exports/dashboard_usage.py]: this can be simplified down to
```
state_name = self.get_location_name_from_id(record.state_id)
district_name = ''
block_name = ''

if record.user_level >1:
    district_name = self.get_location_name_from_id(record.district_id)
if record.user_level >2:
    block_name = self.get_location_name_from_id(record.block_id)

```

File: custom/icds_reports/sqldata/exports/dashboard_usage.py
Comment (Rohit25negi) [inline custom/icds_reports/sqldata/exports/dashboard_usage.py]: can you use 'india_now' instead to ensure its comparing based on IST time.

File: custom/icds_reports/sqldata/exports/dashboard_usage.py
Comment (Rohit25negi) [inline custom/icds_reports/sqldata/exports/dashboard_usage.py]: Also this date returned from activity report is in IST you will have convert it to IST.

File: custom/icds_reports/sqldata/exports/dashboard_usage.py
Comment (Rohit25negi) [inline custom/icds_reports/sqldata/exports/dashboard_usage.py]: same comment. This last_activity will be in UTC.

File: custom/icds_reports/sqldata/exports/dashboard_usage.py
Comment (Rohit25negi) [inline custom/icds_reports/sqldata/exports/dashboard_usage.py]: you can filter just the `aggregation_level=3` you don't need to filter on other level.s

**Review by Rohit25negi — CHANGES_REQUESTED** [review]
This looks way easier to read than before.
left a bunch of comments.

File: custom/icds_reports/sqldata/exports/dashboard_usage.py
Comment (challabeehyv) [reply custom/icds_reports/sqldata/exports/dashboard_usage.py]: this is an instance varibale that is initiated in the constructor

File: custom/icds_reports/sqldata/exports/dashboard_usage.py
Comment (challabeehyv) [reply custom/icds_reports/sqldata/exports/dashboard_usage.py]: yes. Both have the method

File: custom/icds_reports/sqldata/exports/dashboard_usage.py
Comment (challabeehyv) [reply custom/icds_reports/sqldata/exports/dashboard_usage.py]: yeah.makes sense

File: custom/icds_reports/sqldata/exports/dashboard_usage.py
Comment (challabeehyv) [reply custom/icds_reports/sqldata/exports/dashboard_usage.py]: removed

File: custom/icds_reports/sqldata/exports/dashboard_usage.py
Comment (challabeehyv) [reply custom/icds_reports/sqldata/exports/dashboard_usage.py]: changed

File: custom/icds_reports/sqldata/exports/dashboard_usage.py
Comment (challabeehyv) [reply custom/icds_reports/sqldata/exports/dashboard_usage.py]: removed

File: custom/icds_reports/sqldata/exports/dashboard_usage.py
Comment (Rohit25negi) [reply custom/icds_reports/sqldata/exports/dashboard_usage.py]: you could directly create a `@cached_property` like [this](https://github.com/dimagi/commcare-hq/pull/26036/files#diff-313adb3e671e64994dde31ad142a29faR35) and not create seprate function and a property. that would be neat.

File: custom/icds_reports/sqldata/exports/dashboard_usage.py
Comment (Rohit25negi) [reply custom/icds_reports/sqldata/exports/dashboard_usage.py]: What will happen when `record.block_id` is 'All' what will get_location_name_from_id return?

**Review by Rohit25negi — CHANGES_REQUESTED** [review]
left a few questions

File: custom/icds_reports/sqldata/exports/dashboard_usage.py
Comment (challabeehyv) [reply custom/icds_reports/sqldata/exports/dashboard_usage.py]: done

**Review by Rohit25negi — APPROVED** [review]

**Comment by Rohit25negi** [comment]
Do not merge this until its parent branch is merged.

**Comment by Rohit25negi** [comment]
closing and reopening to see if the travis build is trigger or not. I have approved it though.

**Comment by Rohit25negi** [comment]
yay. that wrked.and build passed. merging this.

**Pull request merged**
